src/main/java/org/opensearch/neuralsearch/processor/chunker/DelimiterChunker.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterParser.parseStringParameter;
+import static org.opensearch.neuralsearch.processor.chunker.ChunkerParameterValidator.validateStringParameter;
+
+/**
+ * The implementation {@link Chunker} for delimiter algorithm
+ */
+public class DelimiterChunker implements Chunker {
+
+    public static final String ALGORITHM_NAME = "delimiter";
+    public static final String DELIMITER_FIELD = "delimiter";
+    public static final String DEFAULT_DELIMITER = "\n\n";
+
+    private String delimiter;
+
+    public DelimiterChunker(final Map<String, Object> parameters) {
+        validateParameters(parameters);
+        parseParameters(parameters);
+    }
+
+    /**
+     * Validate the parameters for delimiter algorithm,
+     * will throw IllegalArgumentException if delimiter is not a string or empty
+     *
+     * @param parameters a map containing parameters, containing the following parameters
+     * 1. A string as the paragraph split indicator
+     */
+    @Override
+    public void validateParameters(Map<String, Object> parameters) {
+        validateStringParameter(parameters, DELIMITER_FIELD, false);
+    }
+
+    /**
+     * Parse the parameters for delimiter algorithm,
+     * will throw IllegalArgumentException if delimiter is not a string or empty
+     *
+     * @param parameters a map containing parameters, containing the following parameters
+     * 1. A string as the paragraph split indicator
+     */
+    @Override
+    public void parseParameters(Map<String, Object> parameters) {
+        this.delimiter = parseStringParameter(parameters, DELIMITER_FIELD, DEFAULT_DELIMITER);
+    }
+
+    /**
+     * Return the chunked passages for fixed token length algorithm
+     *
+     * @param content input string
+     * @param runtimeParameters a map for runtime parameters, but not needed by delimiter algorithm
+     */
+    @Override
+    public List<String> chunk(final String content, final Map<String, Object> runtimeParameters) {
+        List<String> chunkResult = new ArrayList<>();
+        int start = 0, end;
+        int nextDelimiterPosition = content.indexOf(delimiter);
+
+        while (nextDelimiterPosition != -1) {
```

Comment:
> we should add another exit condition here where after some `X` number of chunks we will not chunk the string. If we keep this code like this it is open for abuse.

Good point! That's why we introduce a parameter `max_chunk_limit` to restrict the number of chunks for a single ingest document.

Reply:
> Another thing is these chunks will be sent to create embeddings going forward which will become internal documents of lucene. If string contains a lot of delimiters then it can lead of ingestion failures where you are not able to ingest documents because you hit the limits of Opensearch and wastage of resources to convert a string to embeddings.

User can specify parameter `max_chunk_limit` to avoid this issue.

Reply:
that relies on user's good intention, how about opposite. this can be flagged at security review as potential breach for dos attack

Reply:
I get your point. I will assign a conservative default value to parameter `max_chunk_limit`. Users will no longer need to assign this parameter in advance.

Reply:
I have introduced a runtime variable for `max_chunk_limit` so that this parameter is effective in the chunking algorithm. The benefit of doing so is that we can return an exception to the user immediately when the size of chunked passage exceeds the chunking limit.